### ISSUE TITLE
Distinction between pump() and pumpUntilNoTransientCallbacks() is not clear

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#8186]
**Case A**

```
await scroll(tester, const Offset(x, y));
await tester.pump();                          // Not sufficient
await tester.pumpUntilNoTransientCallbacks(); // Required
```

**Case B**

```
Scrollable.ensureVisible(tester.element(finder));
await tester.pump();                          // Required
await tester.pumpUntilNoTransientCallbacks(); // Does nothing
```

Most of the time, what you want as a test author is a "pump until the app is stable" method.  What we don't want is for test authors to get in the habit of "I always just do both `pump()` and `pumpUntilNoTransientCallbacks()`, and it seems to work."

**Comment by Hixie** [comment]
@abarth agrees with you. :-)

**Comment by tvolkert** [comment]
Fixed in #8751
